internal/test/issues/issue-1530/issue1530.gen.go

```diff
@@ -0,0 +1,141 @@
+// Package param provides primitives to interact with the openapi HTTP API.
+//
+// Code generated by github.com/deepmap/oapi-codegen/v2 version v2.0.0-00010101000000-000000000000 DO NOT EDIT.
+package param
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/oapi-codegen/runtime"
+)
+
+// Defines values for CatBreed.
+const (
+	MaineCoon CatBreed = "maine_coon"
+	Ragdoll   CatBreed = "ragdoll"
+)
+
+// Defines values for DogBreed.
+const (
+	Beagle DogBreed = "beagle"
+	Poodle DogBreed = "poodle"
+)
+
+// Cat defines model for cat.
+type Cat struct {
+	Breed CatBreed `json:"breed"`
+}
+
+// CatBreed defines model for Cat.Breed.
+type CatBreed string
+
+// Dog defines model for dog.
+type Dog struct {
+	Breed DogBreed `json:"breed"`
+}
+
+// DogBreed defines model for Dog.Breed.
+type DogBreed string
+
+// Pet defines model for pet.
+type Pet struct {
+	union json.RawMessage
+}
+
+// AsDog returns the union data inside the Pet as a Dog
+func (t Pet) AsDog() (Dog, error) {
+	var body Dog
+	err := json.Unmarshal(t.union, &body)
+	return body, err
+}
+
+// FromDog overwrites any union data inside the Pet as the provided Dog
+func (t *Pet) FromDog(v Dog) error {
+	v.Breed = "beagle"
+	v.Breed = "poodle"
+	b, err := json.Marshal(v)
+	t.union = b
+	return err
+}
+
+// MergeDog performs a merge with any union data inside the Pet, using the provided Dog
+func (t *Pet) MergeDog(v Dog) error {
+	v.Breed = "beagle"
+	v.Breed = "poodle"
+	b, err := json.Marshal(v)
+	if err != nil {
+		return err
+	}
+
+	merged, err := runtime.JSONMerge(t.union, b)
+	t.union = merged
+	return err
+}
+
+// AsCat returns the union data inside the Pet as a Cat
+func (t Pet) AsCat() (Cat, error) {
+	var body Cat
+	err := json.Unmarshal(t.union, &body)
+	return body, err
+}
+
+// FromCat overwrites any union data inside the Pet as the provided Cat
+func (t *Pet) FromCat(v Cat) error {
+	v.Breed = "maine_coon"
+	v.Breed = "ragdoll"
```

Comment:
This PR goes a long way to fixing the issue for me, but it doesn't address this part of the codegen

Reply:
I've submitted an enhanced version of this PR -- https://github.com/oapi-codegen/oapi-codegen/pull/1789 -- which I think should address this issue